### PR TITLE
feature(menus): allow users to access their settings from the hover menu

### PR DESCRIPTION
**Pull request by hypeJunction**

### CODE REVIEW

File: engine/lib/users.php [hunk 1]
```diff
@@ -481,6 +481,11 @@ function elgg_user_hover_menu($hook, $type, $return, $params) {
 		$item = new \ElggMenuItem('avatar:edit', elgg_echo('avatar:edit'), $url);
 		$item->setSection('action');
 		$return[] = $item;
+
+		$url = "settings/user/$user->username";
+		$item = new \ElggMenuItem('settings:edit', elgg_echo('settings:edit'), $url);
+		$item->setSection('action');
+		$return[] = $item;
```
Comment (mrclay) [inline engine/lib/users.php]: It looks like admins will see two menu items for this (other in admin section). Should we make a 3rd case (if is current user OR is admin, show Edit Settings)?

Comment (hypeJunction) [inline engine/lib/users.php]: Only if viewing other admin's hover menu. I would say let's add $user->canEdit() section, and move all the stuff there, except for truly admin stuff.

Comment (jdalsem) [inline engine/lib/users.php]: > I would say let's add $user->canEdit() section, and move all the stuff there, except for truly admin stuff.

:+1:

**Comment by jdalsem** [comment]
@hypeJunction why was this closed?

**Comment by hypeJunction** [comment]
No more interest in pursuing this

**Pull request closed without merging**